base/src/iosMain/kotlin/VersionUtils.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ Copyright (c) 2020. Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base
+
+import platform.UIKit.UIDevice
+
+data class IOSVersion(val major: Int, val minor: Int, val patch: Int) {
+
+    companion object {
+        val systemVersion: IOSVersion
+            get() {
+                val versions = UIDevice.currentDevice.systemVersion.split(".").map { it.toIntOrNull() ?: 0 }
+                return IOSVersion(versions.getOrNull(0) ?: 0, versions.getOrNull(1) ?: 0, versions.getOrNull(2) ?: 0)
+            }
+    }
+
+    fun isOSVersionOrNewer(version: IOSVersion): Boolean {
+        return if (this.major > version.major) {
```

Comment:
In my opinion it's not readable, I like more classic way, `if` and inside block `return true`

Reply:
For me looks good - more like functional programming and in kotlin it is quite common, but in a multiplatform project it probably makes sense to avoid such constructions

Reply:
It's idiomatic Kotlin the way it's written IMHO.

Reply:
This can be more readable with using `when`